foundation/org.eclipse.persistence.core/src/main/java/module-info.java

```diff
@@ -38,6 +38,8 @@
     requires static jakarta.messaging;
     requires static jakarta.resource;
 
+    uses org.eclipse.persistence.internal.databaseaccess.spi.JsonPlatformProvider;
```

Comment:
so the core module will find it; what about eclipselink.jar?

Reply:
Let me guess ...it won't find it.